pkg/infrastructure/azure/azure.go

```diff
@@ -339,3 +344,49 @@ func randomString(length int) string {
 
 	return string(s)
 }
+
+// Ignition provisions the Azure container that holds the bootstrap ignition
+// file.
+func (p Provider) Ignition(ctx context.Context, in clusterapi.IgnitionInput) ([]byte, error) {
+	session, err := in.InstallConfig.Azure.Session()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get session: %w", err)
+	}
+
+	bootstrapIgnData := in.BootstrapIgnData
+	subscriptionID := session.Credentials.SubscriptionID
+	cloudConfiguration := session.CloudConfig
+
+	ignitionContainerName := "ignition"
+	blobName := "bootstrap.ign"
+	blobURL := fmt.Sprintf("%s/%s/%s", p.StorageURL, ignitionContainerName, blobName)
+
+	// Create ignition blob storage container
+	createBlobContainerOutput, err := CreateBlobContainer(ctx, &CreateBlobContainerInput{
+		ContainerName:        ignitionContainerName,
+		SubscriptionID:       subscriptionID,
+		ResourceGroupName:    p.ResourceGroupName,
+		StorageAccountName:   p.StorageAccountName,
+		StorageClientFactory: p.StorageClientFactory,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	blobIgnitionContainer := createBlobContainerOutput.BlobContainer
+	logrus.Debugf("BlobIgnitionContainer.ID=%s", *blobIgnitionContainer.ID)
+
+	_, err = CreateBlockBlob(ctx, &CreateBlockBlobInput{
```

Comment:
in terraform we are switching between block & page depending on customer-managed keys: https://github.com/openshift/installer/blob/master/data/data/azure/bootstrap/main.tf#L68

I'm guessing that only block types supports customer-managed keys. So we need to use it for that case, and perhaps it is fine to always use block and we don't need to use page like terraform does?

Reply:
I am writing this based on how we do this in Terraform. If a keyvault is specified, we use a page blob, if not, we use block blob. At this point, this is a WIP and I am only interested in block blobs at the moment. When I can get this functional, I will add the logic for when a keyvault is set.

Reply:
I think just block blobs can be used here. I remember terraform enforcing the use of Premium tier storage account that did not have support for block blobs. I see a few types in Azure like the `Premium Block blob` that might support customer-managed keys. Need to investigate.
https://learn.microsoft.com/en-us/azure/storage/common/storage-account-overview#types-of-storage-accounts

Reply:
I am not clear on what is being asked here. As-is, this is creating a block blob, the same as it currently does using Terraform: https://github.com/openshift/installer/blob/39ba56297d995b1f3e51ef86f5e7c8aa4436316c/data/data/azure/bootstrap/main.tf#L68

Are you guys asking me to change this to something else?